### ISSUE TITLE
Schema doesn't allow empty columns

### ISSUE THREAD

**Issue by andrewleader** [microsoft/AdaptiveCards#332]
A common thing developers do is use columns to get a specific percent-based width on images as seen in the example below.

![image](https://cloud.githubusercontent.com/assets/13246069/26079427/9445aa0a-3977-11e7-8c0c-018b6c98e95c.png)

The schema requires that a column have an items array... and additionally the UWP renderer requires that the array have an item... the Web renderer also requires that the array has an item, but even worse, it can't be a blank TextBlock... it has to have a space in it, like `"text": " "` rather than just `"text": ""`

Ideally, you should just be able to use blank columns as seen below...

```json
        {
          "type": "ColumnSet",
          "columns": [
            {
              "type": "Column",
              "size": 1
            },
            {
              "type": "Column",
              "size": 2,
              "items": [
                {
                  "type": "Image",
                  "url": "https://unsplash.it/64?image=1027",
                  "size": "stretch",
                  "style": "person"
                }
              ]
            },
            {
              "type": "Column",
              "size": 1
            }
          ]
        },
```

But instead you have to do

```json
        {
          "type": "ColumnSet",
          "columns": [
            {
              "type": "Column",
              "size": 1,
              "items": [
                {
                  "type": "TextBlock",
                  "text":  " "
                }
              ]
            },
            {
              "type": "Column",
              "size": 2,
              "items": [
                {
                  "type": "Image",
                  "url": "https://unsplash.it/64?image=1027",
                  "size": "stretch",
                  "style": "person"
                }
              ]
            },
            {
              "type": "Column",
              "size": 1,
              "items": [
                {
                  "type": "TextBlock",
                  "text": " "
                }
              ]
            }
          ]
        },
```

Using the size property on the image isn't an option if you need a specific percent-width. This is a regression from what was possible in Live Tiles and toasts.

**Comment by dclaux** [comment]
Is the ask to make it possible to size an image as a percentage of the available width in the container it's in?

**Comment by dclaux** [comment]
I have updated the HTML renderer so it allows empty columns. We still have to discuss the schema and whether we should update it.

**Comment by anbare** [comment]
Sizing the image as a percent of available width would certainly be the much more elegant solution to this case :) But there could probably be other cases where there's both text and an image that needs to be in the center, there might be cases where empty columns are still useful.

In general I feel like empty columns probably shouldn't break the payload, seems like an odd thing to break on.

**Comment by andrewleader** [comment]
Closing as we implemented this sometime in 1.0/1.1 (items is no longer required)
